pootle/apps/pootle_project/models.py

```diff
@@ -420,6 +423,13 @@ def delete(self, *args, **kwargs):
 
         directory.delete()
 
+    def clean(self):
+        if not self.code:
+            raise ValidationError('A code must be provided.')
```

Comment:
It turns out that `blank=False` is enforced if in `save()` the `full_clean()` method is called.